test/transcoding/FPGAIVDepLoopAttr.ll

```diff
@@ -0,0 +1,416 @@
+  store i32 %61, i32* %59, align 4, !tbaa !9
+  br label %62
+
+62:                                               ; preds = %49
+  %63 = load i32, i32* %6, align 4, !tbaa !9
+  %64 = add nsw i32 %63, 1
+  store i32 %64, i32* %6, align 4, !tbaa !9
+  ; CHECK-LLVM: br label %{{.*}}, !llvm.loop ![[EMB_MD_LOOP_DIM_3:[0-9]+]]
+  br label %44, !llvm.loop !21
+
+65:                                               ; preds = %47
+  br label %66
+
+66:                                               ; preds = %65
+  %67 = load i32, i32* %5, align 4, !tbaa !9
+  %68 = add nsw i32 %67, 1
+  store i32 %68, i32* %5, align 4, !tbaa !9
+  ; CHECK-LLVM: br label %{{.*}}, !llvm.loop ![[EMB_MD_LOOP_DIM_2:[0-9]+]]
+  br label %25, !llvm.loop !23
+
+69:                                               ; preds = %28
+  br label %70
+
+70:                                               ; preds = %69
+  %71 = load i32, i32* %3, align 4, !tbaa !9
+  %72 = add nsw i32 %71, 1
+  store i32 %72, i32* %3, align 4, !tbaa !9
+  ; CHECK-LLVM: br label %{{.*}}, !llvm.loop ![[EMB_MD_LOOP_DIM_1:[0-9]+]]
+  br label %10, !llvm.loop !25
+
+73:                                               ; preds = %13
+  %74 = bitcast [10 x i32]* %2 to i8*
+  call void @llvm.lifetime.end.p0i8(i64 40, i8* %74) #4
+  %75 = bitcast [10 x i32]* %1 to i8*
+  call void @llvm.lifetime.end.p0i8(i64 40, i8* %75) #4
+  ret void
+}
+
+; CHECK-SPIRV: Function
+; Function Attrs: nounwind
+define dso_local spir_func void @_Z27ivdep_mul_arrays_and_globalv() #3 {
+  ; CHECK-SPIRV: Variable [[TYPE_PTR]] [[ARRAY_A:[0-9]+]]
+  ; CHECK-LLVM: %[[SIMPLE_ARRAY_A:[0-9]+]] = alloca [10 x i32]
+  %1 = alloca [10 x i32], align 4
+  ; CHECK-SPIRV: Variable [[TYPE_PTR]] [[ARRAY_B:[0-9]+]]
+  ; CHECK-LLVM: %[[SIMPLE_ARRAY_B:[0-9]+]] = alloca [10 x i32]
+  %2 = alloca [10 x i32], align 4
+  ; CHECK-SPIRV: Variable [[TYPE_PTR]] [[ARRAY_C:[0-9]+]]
+  ; CHECK-LLVM: %[[SIMPLE_ARRAY_C:[0-9]+]] = alloca [10 x i32]
+  %3 = alloca [10 x i32], align 4
+  ; CHECK-SPIRV: Variable [[TYPE_PTR]] [[ARRAY_D:[0-9]+]]
+  ; CHECK-LLVM: %[[SIMPLE_ARRAY_D:[0-9]+]] = alloca [10 x i32]
+  %4 = alloca [10 x i32], align 4
+  %5 = alloca i32, align 4
+  %6 = bitcast [10 x i32]* %1 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 40, i8* %6) #4
+  %7 = bitcast [10 x i32]* %2 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 40, i8* %7) #4
+  %8 = bitcast [10 x i32]* %3 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 40, i8* %8) #4
+  %9 = bitcast [10 x i32]* %4 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 40, i8* %9) #4
+  %10 = bitcast i32* %5 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 4, i8* %10) #4
+  store i32 0, i32* %5, align 4, !tbaa !9
+  br label %11
+
+11:                                               ; preds = %29, %0
+  %12 = load i32, i32* %5, align 4, !tbaa !9
+  %13 = icmp ne i32 %12, 10
+  ; CHECK-SPIRV: LoopMerge [[CONTINUE_BLOCK:[0-9]+]] {{[0-9]+}} {{[0-9]+}} 5891 4 [[ARRAY_A]] 5 [[ARRAY_D]] 5 [[ARRAY_B]] 6 [[ARRAY_C]] 0
```

Comment:
Something wrong with LoopMerge. It shall be:
`LoopMerge {{Merge Block}} {{Continue Target}} [[Loop Control = 5891]] ...`
but here I see three numbers before Loop Control, or I'm missing something?

Reply:
It's more so that I've obfuscated the literal of the ExtendedLoopControl under the third number. These lines need to be reworked into:
```suggestion
  ; CHECK-SPIRV: LoopMerge [[MERGE_BLOCK:[0-9]+]] {{[0-9]+}} 2147483648 5891 4 [[ARRAY_A]] 5 [[ARRAY_D]] 5 [[ARRAY_B]] 6 [[ARRAY_C]] 0
```